packages/cfd/src/Containers/cfd-financial-stp-real-account-signup.tsx

```diff
@@ -1,83 +1,78 @@
 import React from 'react';
 import { Div100vhContainer } from '@deriv/components';
 import { isDesktop, getAuthenticationStatusInfo } from '@deriv/shared';
-import { connect } from '../Stores/connect';
-import { LandingCompany, ResidenceList, GetSettings, StatesList, GetAccountStatus } from '@deriv/api-types';
 import CFDPOA from '../Components/cfd-poa';
 import CFDPOI from '../Components/cfd-poi';
 import CFDPersonalDetailsContainer from './cfd-personal-details-container';
-import RootStore from '../Stores/index';
-
-type TAuthenticationStatus = { document_status: string; identity_status: string };
-
-type TStoreProofOfAddressArgs = {
-    file_uploader_ref: HTMLDivElement | null;
-    values: { [key: string]: string };
-};
-
-type TRemoveNotificationMessage = {
-    key: string;
-    should_show_again: boolean;
-};
-
-type TGetSettings = GetSettings & {
-    upload_file?: string;
-    poi_state?: string;
-};
+import { observer, useStore } from '@deriv/stores';
+import { useCfdStore } from '../Stores/Modules/CFD/Helpers/useCfdStores';
 
 type TCFDFinancialStpRealAccountSignupProps = {
-    addNotificationByKey: (key: string) => void;
-    authentication_status: TAuthenticationStatus;
-    get_settings: TGetSettings;
-    client_email: string;
-    context: RootStore;
-    is_fully_authenticated: boolean;
-    landing_company: LandingCompany;
-    refreshNotifications: () => void;
-    removeNotificationMessage: () => void;
-    removeNotificationByKey: (args: TRemoveNotificationMessage) => void;
-    residence_list: ResidenceList;
-    states_list: StatesList;
-    storeProofOfAddress: TStoreProofOfAddressArgs;
-    fetchStatesList: () => void;
-    account_status: GetAccountStatus;
     onFinish: () => void;
-    jurisdiction_selected_shortcode: string;
-    has_submitted_cfd_personal_details: boolean;
 };
 
 type TNextStep = (index: number, value: { [key: string]: string | undefined }) => void;
 
-type TItemsState = {
+type TItemsState<T> = {
```

Comment:
It would be better to restrict the range of supported types for <T>, it is expecting everything this way and this is not a good way to use our generics.

Reply:
```ts
type TItem = {
    refreshNotifications: TCoreStores['notifications']['refreshNotifications'];
    removeNotificationMessage: TCoreStores['notifications']['removeNotificationMessage'];
    removeNotificationByKey: TCoreStores['notifications']['removeNotificationByKey'];
    addNotificationMessageByKey: TCoreStores['notifications']['addNotificationMessageByKey'];
    authentication_status: TCoreStores['client']['authentication_status'];
    account_settings: TCoreStores['client']['account_settings'];
    email: TCoreStores['client']['email'];
    is_fully_authenticated: TCoreStores['client']['is_fully_authenticated'];
    landing_company: TCoreStores['client']['landing_company'];
    residence_list: TCoreStores['client']['residence_list'];
    states_list: TCoreStores['client']['states_list'];
    fetchStatesList: TCoreStores['client']['fetchStatesList'];
    account_status: TCoreStores['client']['account_status'];
    storeProofOfAddress: TCoreStores['modules']['cfd']['storeProofOfAddress'];
    jurisdiction_selected_shortcode: TCoreStores['modules']['cfd']['jurisdiction_selected_shortcode'];
    has_submitted_cfd_personal_details: TCoreStores['modules']['cfd']['has_submitted_cfd_personal_details'];
    onFinish: TCFDFinancialStpRealAccountSignupProps['onFinish'];
};

type TItemsState<T extends TItem> = {
    body: typeof CFDPOI | typeof CFDPOA | typeof CFDPersonalDetailsContainer;
    form_value: { [key: string]: string | undefined };
    forwarded_props: Array<Partial<keyof T>>;
};

```

How about this, does this looks good ?

Reply:
Added the constraint in the generic

@hirad-deriv